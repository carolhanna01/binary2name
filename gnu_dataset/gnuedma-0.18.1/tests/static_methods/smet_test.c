/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <edma.h>

int
main(int argc,char *argv[])
{
  ESint32          res;
  /* Initialize EDMA System*/
  EDMAInit();

  /* Put your C code here */
  edma_smet3 ("UNKNOWN_CLASS","SMet1","Hola Mundo!!!",50);
  edma_printf ("===================================================");
  edma_smet3 ("STATIC_MET_TEST","UnknownMethod","Hola Mundo!!!",50);

  edma_printf ("-> RUnning SMet1");
  edma_smet3 ("STATIC_MET_TEST","SMet1","Hola Mundo!!!",50);
  edma_printf ("-> RUnning SMet2");
  res = (ESint32) edma_smet3 ("STATIC_MET_TEST","SMet2",10,20);
  edma_printf ("Result : %d\n",res);

  /* Shutdown EDMA Suystem */
  EDMAEnd();
  return 0;
}
