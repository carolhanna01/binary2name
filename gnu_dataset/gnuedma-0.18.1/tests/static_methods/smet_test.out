
GNU/EDMA 0.16.4 build [Jan 30 2010][Beta][0.16.4] starting up...
EDMA Startup process... OK
[EDMA_EXCEPTION] <1> :: [edma_met3] Class 'UNKNOWN_CLASS' don't exist
===================================================
[EDMA_EXCEPTION] <1> :: [edma_smet3x] Method 'UnknownMethod' not found in class 'STATIC_MET_TEST'
-> RUnning SMet1
Running Static Method SMet1 of class STATIC_MET_TEST[262]
Par 1: Hola Mundo!!!
Par 2: 50
-> RUnning SMet2
Running Static Method SMet2 of class STATIC_MET_TEST[262]
Par 1: 10
Par 2: 20
sum: 30
Result : 30

---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
